components/custom/survey/CustomSankey.tsx

```diff
@@ -103,20 +92,43 @@ export const CustomSankey: React.FC<SankeyProps> = ({ nodes, links }) => {
 
     // Custom Link Component with Label 
     const CustomLink = (props: CustomLinkProps) => { 
-        const { sourceX, targetX, sourceY, targetY, sourceControlX, targetControlX, linkWidth, payload } = props; 
+        const { 
+            sourceX, 
+            targetX, 
+            sourceY, 
+            targetY, 
+            sourceControlX, 
+            targetControlX, 
+            linkWidth, 
+            payload 
+        } = props; 
+        const sourceColor = payload.source.color || "rgb(var(--survey-grey-mid))";
+        const targetColor = payload.target.color || "rgb(var(--survey-grey-mid))";
+
+        const sanitize = (str: string) => str.replace(/[^a-zA-Z0-9-_]/g, ""); // handle special characters (eg for tenure like "I don't know")
+        const sourceLabel = sanitize(payload.source.name.replace(/\s+/g, "-"));
+        const targetLabel = sanitize(payload.target.name.replace(/\s+/g, "-"));
+        const gradientId = `link-gradient-${sourceLabel}-to-${targetLabel}`; // we need a unique ID
```

Comment:
Not sure it would work here, but there's a built in `useId()` hook for times like this ([docs](https://react.dev/reference/react/useId)).